File: frontend/food-delivery-platform/src/models/responses/accounts/OnboardingLinkResponse.ts
// Reply for GET /account/onboarding/{businessId}.
// 200 → status="ready" with the Stripe-hosted onboarding URL.
// 202 → status="provisioning"; Stripe Connect account isn't ready yet
//       (StripeAccountProvisioningWorker is in flight). Retry after the hint.
export interface OnboardingLinkResponse {
    status: "ready" | "provisioning"
    url?: string | null
    retryAfterSeconds?: number | null
}

File: backend/UserService/DF.UserService.Contracts/Models/Response/OnboardingLinkResponse.cs
namespace DF.UserService.Contracts.Models.Response;

/// <summary>
/// Reply for GET /api/account/onboarding/{businessId}.
/// Status "ready" carries the Stripe-hosted onboarding URL.
/// Status "provisioning" tells the client the Stripe account isn't ready yet
/// (StripeAccountProvisioningWorker still in flight); RetryAfterSeconds is a
/// hint for when to retry.
/// </summary>
public record OnboardingLinkResponse(
    string Status,
    string? Url,
    int? RetryAfterSeconds)
{
    public const string ReadyStatus = "ready";
    public const string ProvisioningStatus = "provisioning";

    public static OnboardingLinkResponse Ready(string url) =>
        new(ReadyStatus, url, null);

    public static OnboardingLinkResponse Provisioning(int retryAfterSeconds = 15) =>
        new(ProvisioningStatus, null, retryAfterSeconds);
}
